support/Pipelines/Bbh/FindHorizon.py

```diff
@@ -0,0 +1,231 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import glob
+import logging
+from pathlib import Path
+from typing import Optional, Sequence, Type, Union
+
+import click
+
+import spectre.IO.H5 as spectre_h5
+from spectre.ApparentHorizonFinder import FastFlow, FlowType, Status
+from spectre.DataStructures import DataVector
+from spectre.DataStructures.Tensor import tnsr
+from spectre.IO.Exporter import interpolate_tensors_to_points
+from spectre.Spectral import Basis, Quadrature
+from spectre.SphericalHarmonics import (
+    Strahlkorper,
+    cartesian_coords,
+    ylm_legend_and_data,
+)
+from spectre.Visualization.OpenVolfiles import open_volfiles_command
+from spectre.Visualization.ReadH5 import list_observations
+
+logger = logging.getLogger(__name__)
+
+
+def _strahlkorper_vol_data(strahlkorper):
+    """Volume data representation of a Strahlkorper, can be written to H5."""
+    coords = cartesian_coords(strahlkorper)
+    return [
+        spectre_h5.ElementVolumeData(
+            element_name="Horizon",
+            components=[
+                spectre_h5.TensorComponent(
+                    "InertialCoordinates_" + "xyz"[d], coords[d]
+                )
+                for d in range(3)
+            ],
+            extents=strahlkorper.physical_extents,
+            basis=2 * [Basis.SphericalHarmonic],
+            quadrature=[
+                Quadrature.Gauss,
+                Quadrature.Equiangular,
+            ],
+        )
+    ]
+
+
+def find_horizon(
+    h5_files: Union[str, Sequence[str]],
+    subfile_name: str,
+    obs_id: int,
+    obs_time: float,
+    l_max: int,
+    initial_radius: float,
+    center: Sequence[float],
+    output: Optional[str] = None,
+    output_coeffs_subfile: Optional[str] = None,
+    output_coords_subfile: Optional[str] = None,
+    tensor_names: Optional[Sequence[str]] = None,
+):
+    """Find an apparent horizon in volume data.
+
+    Arguments:
+      h5_files: List of H5 files containing volume data or glob pattern.
+      subfile_name: Name of the volume data subfile in the 'h5_files'.
+      obs_id: Observation ID in the volume data.
+      obs_time: Time of the observation.
+      l_max: Maximum l-mode for the horizon search.
+      initial_radius: Initial coordinate radius of the horizon.
+      center: Coordinate center of the horizon.
```

Comment:
This is the expansion center of the horizon, not a physical center. Should note that somehow